src/js/components/DataTable/Header.js

```diff
@@ -165,15 +165,20 @@ const Header = forwardRef(
           ...columns.map(({ property }) => cellWidths[property]),
         ]);
       }
-    };
-
-    const updateWidths = (property, width) => {
-      const cellWidths = cellWidthsRef.current;
-      // save width for this column. Subtract 1 to avoid gap due to rounding
-      cellWidths[property] = width - 1;
-      if (timerRef.current) clearTimeout(timerRef.current);
-      timerRef.current = setTimeout(handleWidths, WIDTH_UPDATE_DELAY);
-    };
+    }, [columns, onSelect, onWidths, selected]);
+
+    const updateWidths = useCallback(
+      (property, width) => {
+        if (typeof width !== 'number') return;
+        // Only update if width actually changed
+        if (cellWidthsRef.current[property] !== width) {
```

Comment:
could `cellWidthsRef` ever be undefined? or null?

Reply:
I added a `?` just in case

Reply:
okay!